src/main/java/net/datafaker/service/FakeValuesContext.java

```diff
@@ -89,13 +89,23 @@ public URL getUrl() {
     @Override
     public boolean equals(Object o) {
         if (this == o) return true;
-        if (!(o instanceof FakeValuesContext that)) return false;
-        return sLocale == that.sLocale && Objects.equals(filename, that.filename) && Objects.equals(path, that.path) && Objects.equals(url, that.url);
+        if (o == null || getClass() != o.getClass()) return false;
+
+        FakeValuesContext that = (FakeValuesContext) o;
+
+        if (!Objects.equals(sLocale, that.sLocale)) return false;
+        if (!Objects.equals(filename, that.filename)) return false;
+        if (!Objects.equals(path, that.path)) return false;
+        return Objects.equals(url, that.url);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(sLocale, filename, path, url);
+        int result = sLocale != null ? sLocale.hashCode() : 0;
+        result = 31 * result + (filename != null ? filename.hashCode() : 0);
+        result = 31 * result + (path != null ? path.hashCode() : 0);
+        result = 31 * result + (url != null ? url.hashCode() : 0);
```

Comment:
Apparently qudona and other SAST suggest against this with URLs because it actually causes a DNS resolution, which can be slow or fail. 

> In general it is better to use the URI class until access to the resource is actually needed, at which point you can just convert the URI to a URL using URI.toURL().

Reply:
hmmm, i wonder why qudona and other SAST keep silence when we invoke `Objects.hash(args)` and then under the hood `Arrays.hashcode(args)` and the same code as above....

Reply:
🤷‍♂️ it’s just the URL one in particular because of the impl I guess

Reply:
ok, anyway changed it for url
+ added hashes as constants for constant fields since there is no need to calculate them each time
may be similar things could be done for other classes however in a separate PR + need some measurements (for this one it improves a bit about 3-4 %)